.github/workflows/pr-tests.yml

```
@@ -0,0 +1,208 @@
+          git fetch origin "${{ github.base_ref }}" --depth=1
+          CHANGED=$(git diff --name-only --diff-filter=AMRT "origin/${{ github.base_ref }}"... | grep -E '\\.py$' || true)
+          printf "Changed Python files:\n%s\n" "$CHANGED"
+          {
+            echo "files<<EOF"
+            printf '%s\n' "$CHANGED"
+            echo "EOF"
+          } >> "$GITHUB_OUTPUT"
+
+      - name: Mypy (changed files only)
+        if: github.event_name != 'pull_request' || steps.changed.outputs.files != ''
+        run: |
```

actionlint (1.7.8)

shellcheck reported issue in this script: SC1009:info:1:53: The mentioned syntax error was in this then clause

(shellcheck)